pl_examples/hydra_examples/conf/optimizer.py

```diff
@@ -0,0 +1,108 @@
+from dataclasses import dataclass
```

Comment:
I think the dataclasses for optimizer and scheduler should probably be a part of PL. It makes more sense than evey user re-defining Adam, AdamW, CosineConf etc.

Speaking of which, we should be consistent. Why Either consistently add Conf suffic or consistently not add it.

Reply:
1.) IMO you're right, this is not specific to any research area so this should be part of lightning if we decide to move major things from this pr to lightning core

Reply:
I'd like to add that in my present use cases, although I don't necessarily want to use the structured configs paradigm for everything at the moment as I'm kind of enjoying the simplicity of a terse yaml heirarchy, I _really_ like the idea of supporting `trainer_conf.py`, `optimizer.py`, and `scheduler.py` via the structured configs. 

It feels very similar in spirit to maintaining `Trainer`'s static method `add_argparse_args()`. On that note, maybe I missed this in the earlier discussion, but what are the actual cons to including `optimizer_conf.py` and `scheduler_conf.py` in PTL? Perhaps aggregating them into a singular `hydra_conf.py` in core so there's only one file to maintain?

Reply:
@romesc-CMU: 
As a user you can still use yaml config files and they will be [validated automatically](https://hydra.cc/docs/next/tutorials/structured_config/schema) against the dataclasses as long as their name and config group matches.